libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -175,6 +176,101 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT __type_always
   using __call _LIBCUDACXX_NODEBUG_TYPE = _Ty;
 };
 
+//! \brief Perform a logical AND operation on a list of Boolean types.
+//!
+//! \note The AND operation is not short-circuiting.
+struct _CCCL_TYPE_VISIBILITY_DEFAULT __type_strict_and
+{
+#  ifndef _LIBCUDACXX_HAS_NO_FOLD_EXPRESSIONS
+  template <class... _Ts>
+  using __call _LIBCUDACXX_NODEBUG_TYPE = bool_constant<(_Ts::value && ...)>;
+#  else
+  template <class... _Ts>
+  using __call _LIBCUDACXX_NODEBUG_TYPE = bool_constant<_CCCL_TRAIT(
+    is_same, integer_sequence<bool, true, _Ts::value...>, integer_sequence<bool, _Ts::value..., true>)>;
+#  endif
```

Comment:
Shouldnt this simply be
```suggestion
  template <class... _Ts>
  using __call _LIBCUDACXX_NODEBUG_TYPE = __fold_and<_Ts::value...>;
```